src/dune_pkg/mount.ml

```diff
@@ -23,6 +23,29 @@ let of_opam_url loc url =
       >>| User_error.ok_exn
     in
     Git rev
+  | `Tar ->
+    let dir = Path.Build.(L.relative root [ "_fetch"; "url" ]) |> Path.build in
```

Comment:
Why do we put this into _fetch? Is this to attempt to make the build system to use this?

Reply:
Yes, exactly. Is there a better to do it?

Reply:
I try to respect the structure for Dune so it doesn't have to redo the work twice. Isn't it supposed to consider it?

Reply:
I don't think that will work. Dune writes a metadata file describing its state of the workspace local cache. If you didn't update this file, it will just delete what you have put in there.

Reply:
But isn't it enough to keep it there temporarily to prevent `dune pkg lock && dune build` from downloading the same tarball twice? With a temporary directory `pkg lock` would download the tarball there once and then `build` would download the tarball again, since it does not know about the previous temporary directory.

Reply:
Did you try to see if it works? I don't think it will because dune needs an internal data structure updated to recognize anything in the _build directory (see `Rule_cache.Workspace_local`).

It would be nice to have it work the way you're saying, but I think you need to fetch the action to be aware of this bag of already downloaded files you can draw from.